.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,38 @@
+name: pull-request
+
+on:
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+
+jobs:
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Prettify code
+        run: npx prettier . --write
+
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v9
+        with:
+          message: 'gha(prettier): format files'
```

Comment:
```suggestion
      - name: Prettify code
        run: npx prettier . --check
```

Reply:
Per the name change suggestion above. Happy to debate this too though.

This comes from my general usage of github actions in that the usage i've seen is either static analysis (linting/type checking) or it's used for more broader CI/CD things (causing net effects on other repos etc). For me, it feels like commiting on top of peoples work can feel a little black magic-y  (for people that are new to the repo), and it causes some more work for the developer if they're wanting to keep their commits clean.

Thoughts?

Reply:
@thisislawatts Maybe you have opinions here too?

Reply:
With branch protection, I'm happy to remove the git commit/push